### ISSUE TITLE
realodData() on picker how?

### ISSUE THREAD

**Issue by reecreateeduardo** [kmcgill88/McPicker-iOS#38]
Hi, since this doesn't require your typical picker, how can I reload my data?

I want to populate my data from firebase, but I can't update my array since there's no way to reloadData on an array.

Thanks :)

**Comment by kmcgill88** [comment]
`McPicker` is currently setup to know the data set before being shown. If the data set changes once McPicker is visible there is no way to `reloadData()`. You would have to dismiss McPicker and `show` it again with the new data.

Is there a reason you can't load from firebase before `show`ing McPicker? Could you elaborate your use-case a little more? ie. Selection of something in component 0 causes data to change in component 1?

**Comment by kmcgill88** [comment]
No response, closing.

**Comment by elishaterada** [comment]
@kmcgill88 I'm creating a custom date picker with an option to select a year or not (hence custom solution vs the native date picker). I'd like to update the 2nd column of dates based on the month and year selected, hence the need for something that reloads the second column list. Any idea on how to achieve this? Thank you.

![Screen Shot 2019-12-24 at 7 50 16 PM](https://user-images.githubusercontent.com/1435061/71431445-c6bd7680-2686-11ea-888f-5614c12585ce.png)

**Comment by elishaterada** [comment]
If anybody is trying to solve the same problem in the future, I solved this by setting a new index for the day using "pickerSelectRowsForComponents".

For example, when the user picks "Feb" and "31", I set the second column index to the max possible day for the February for a given year (or fallback to 1804 or something). Using animation, user will see that they cannot select Feb 31st, vs refreshing entire list.
